poc_slope.py

```diff
@@ -0,0 +1,98 @@
+from numba import njit, float64
+import numpy as np
+from numpy.linalg import norm
+from skglm.solvers import FISTA
+from skglm.datafits import Quadratic
+from skglm.penalties import BasePenalty
+from skglm.estimators import Lasso
+from skglm.utils import make_correlated_data, compiled_clone
+
+
+X, y, _ = make_correlated_data(n_samples=200, n_features=100, random_state=24)
+
+n_samples, n_features = X.shape
+alpha_max = norm(X.T @ y, ord=np.inf) / n_samples
+
+alpha = np.repeat(alpha_max / 10, n_features)
+
+max_iter = 1000
+obj_freq = 100
+tol = 1e-10
+
+
+class SLOPE(BasePenalty):
+    def __init__(self, alphas):
+        self.alphas = alphas
+    
+    def get_spec(self):
+        spec = (
+            ('alphas', float64[:]),
+        )
+        return spec
+    
+    def params_to_dict(self):
+        return dict(alphas=self.alphas)
+
+    def value(self, w):
+        sorted_indices = np.argsort(w)[::-1]  # descending order
+        return np.sum(w * self.alphas[sorted_indices])
+
+    def prox_vec(self, x, stepsize):
+        def _prox(_x, _alphas):
+            sign_x = np.sign(_x)
+            _x = np.abs(_x)
+            sorted_indices = np.argsort(_x)[::-1]
+            prox = fast_prox_SL1(_x[sorted_indices], _alphas)
+            prox[sorted_indices] = prox
+            return prox * sign_x
+        return _prox(x, self.alphas * stepsize)
+
```

Comment:
maybe add a coordinatewise prox that throws an error if called, to make the design robust ?